app/src/main/java/io/github/trojan_gfw/igniter/servers/fragment/ServerListAdapter.java

```diff
@@ -43,6 +43,7 @@ public void replaceData(List<TrojanConfig> data) {
     public void removeItemOnPosition(int pos) {
         mData.remove(pos);
         notifyItemRemoved(pos);
+        notifyDataSetChanged();
```

Comment:
Extra calling of `notifyDataSetChanged()` may not be the proper way to fix the bug. Since `notifyDataSetChanged()` will request full layout in `RecyclerView`. This is a waste of system resources.
 The main cause of the exception is that `getBindingAdapterPosition()` returns `RecyclerView.NO_POSITION` when user clicks delete button while `RecyclerView` is computing layout. Thus validating the position may be the insured way to avoid the exception, like this:
```java
itemView.findViewById(R.id.deleteServerBtn).setOnClickListener(new View.OnClickListener() {
    @Override
    public void onClick(View v) {
        int position = getBindingAdapterPosition();
        if (position != RecyclerView.NO_POSITION && itemClickListener != null) {
            itemClickListener.onItemDelete(mConfig, position);
        }
    }
});
```